src/file/file.controller.ts

```diff
@@ -103,14 +103,17 @@ export class FileController {
   async uploadFile(
     @Query('path') file: string,
     @Body() raw: string,
+    @Res({ passthrough: true }) res: FastifyReply,
   ): Promise<void> {
     try {
       if (typeof raw === 'string' || Buffer.isBuffer(raw)) {
         await fs.promises.access(path.dirname(file), W_OK);
         await fs.promises.writeFile(file, raw);
+        res.status(HttpStatus.OK).send('File uploaded successfully');
       }
     } catch (err) {
       this.logger.error(err.message);
+      res.status(HttpStatus.INTERNAL_SERVER_ERROR).send(err.message);
     }
```

Comment:
It is not clear to me, why did you decide to use `FastifyReply` to do that
```suggestion
    try {
      if (typeof raw === 'string' || Buffer.isBuffer(raw)) {
        await fs.promises.access(path.dirname(file), W_OK);
        await fs.promises.writeFile(file, raw);
        return 'File uploaded successfully';
      }
    } catch (err) {
      this.logger.error(err.message);
      throw err;
    }
```

Reply:
Looked at the method below in this file :)